llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -563,3 +602,14 @@ AIEHazardRecognizer::getSelectedAltOpcode(MachineInstr *MI) const {
     return It->second;
   return std::nullopt;
 }
+
+unsigned AIEHazardRecognizer::getMemoryBanks(MachineInstr *MI) const {
+  unsigned MemoryBankUsed = 0;
```

Comment:
This is a hot function in the scheduler. Perhaps we could first check mayLoad/mayStore for an early return.